File: src/main/java/com/exchange/obs/IOrderBook.java
package com.exchange.obs;

import com.exchange.obs.domain.Price;
import com.exchange.obs.domain.Side;

public interface IOrderBook {

    public void addOrder(Order order);

    public void removeOrder(Order order);

    public void modifyOrder(Order order);

    public void getOrders(char side);

    public Price getLevelPrice(Side side, int level);

    public void displayOrderBook();
}

File: src/main/java/com/exchange/obs/IOrderBook.kt
package com.exchange.obs;

import com.exchange.obs.domain.Price;
import com.exchange.obs.domain.Side;

public interface IOrderBook {

    void addOrder(Order order);

    void removeOrder(Order order);

    void modifyOrder(Order order);

    void getOrders(char side);

    Price getLevelPrice(Side side, int level);

    void displayOrderBook();
}
